### PR TITLE
[parse] adding short disconnect msg support

### PR DESCRIPTION
**Pull request by dscso**
#291

### CODE REVIEW

**Comment by dscso** [comment]
Idk the second commit of the PR is much less important, but in a very data limited environment this could make sense if you are reopening the MQTT connection very often

**Comment by TheNeikos** [comment]
Ah good catch!

There is similar logic for this like this in the repo:

https://github.com/TheNeikos/cloudmqtt/blob/fdcfb587d4ce0ea6ff289de66ab1169554758871/mqtt-format/src/v5/packets/puback.rs#L58-L68

Do you think you could adapt it to this?

**Comment by dscso** [comment]
true, looks better :)

File: mqtt-format/src/v5/packets/mod.rs [hunk 1]
```diff
@@ -162,6 +163,14 @@ impl<'i> MqttPacket<'i> {
...
                 };
                 fixed_header.write(buffer)?;
+                // if reason code is NormalDisconnection AND properties are empty, we can skip writing the payload
+                if p.reason_code == DisconnectReasonCode::NormalDisconnection
+                    && p.properties == DisconnectProperties::new()
+                {
+                    // writing length of packet 0
+                    crate::v5::integers::write_variable_u32(buffer, 0)?;
+                    return Ok(());
+                }
```
Comment (TheNeikos) [inline mqtt-format/src/v5/packets/mod.rs:173]: Could you move this to the `write` of the disconnect packet? Similarly for the binary size.

Comment (dscso) [reply mqtt-format/src/v5/packets/mod.rs]: Thanks for your review! I don't understand the structure of the code yet :)

**Review by TheNeikos — APPROVED** [review]

**Comment by TheNeikos** [comment]
Thank you!

**Pull request merged**